src/main/java/com/google/jenkins/plugins/credentials/oauth/GoogleRobotMetadataCredentials.java

```diff
@@ -46,16 +48,37 @@
 @NameWith(value = GoogleRobotNameProvider.class, priority = 50)
 public final class GoogleRobotMetadataCredentials extends GoogleRobotCredentials
     implements DomainRestrictedCredentials {
+
   /**
-   * Construct a set of service account credentials.
+   * Construct a set of service account credentials with a specific id. It helps for updating
+   * credentials, as well as for migrating old credentials that had no id and relied on the project
+   * id.
    *
+   * @param id the id to assign
    * @param projectId The Pantheon project id associated with this service account
    * @param module The module for instantiating dependent objects, or null.
    */
   @DataBoundConstructor
   public GoogleRobotMetadataCredentials(
-      String projectId, @Nullable GoogleRobotMetadataCredentialsModule module) throws Exception {
-    super(projectId, module);
+      @CheckForNull CredentialsScope scope,
+      String id,
+      String projectId,
+      @Nullable GoogleRobotMetadataCredentialsModule module)
+      throws Exception {
+    super(scope, id, projectId, module);
+  }
+
+  @SuppressFBWarnings(
+      value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE",
+      justification =
+          "for migrating older credentials that did not have a separate id field, and would really "
+              + "have a null id when attempted to deserialize. readResolve overwrites these nulls")
+  private Object readResolve() throws Exception {
+    return new GoogleRobotMetadataCredentials(
+        getCredentialsScope() == null ? CredentialsScope.GLOBAL : getCredentialsScope(),
```

Comment:
I'm not sure how `readResolve` is used? Is it for migration only? If yes, we can simply use `Credentials.GLOBAL`

Reply:
Discussed. The logic should stay to not rewrite scope on each restart.